src/complex/Utilities/Parsing/HDF5/H5.hpp

```diff
@@ -12,8 +26,20 @@ using IdType = int64_t;
 using ErrorType = int32_t;
 using SizeType = unsigned long long;
 
-inline const std::string DataTypeTag = "DataType";
-inline const std::string CompDims = "ComponentDimensions";
-inline const std::string TupleDims = "TupleDimensions";
+static constexpr StringLiteral k_DataTypeTag = "DataType";
+
+static constexpr StringLiteral k_DataStoreTag = "DataStore";
+static constexpr StringLiteral k_TupleShapeTag = "TupleShape";
+static constexpr StringLiteral k_ComponentShapeTag = "ComponentShape";
+static constexpr StringLiteral k_DataObjectIdTag = "DataObjectId";
+static constexpr StringLiteral k_DataArrayTag = "DataArray";
+
+static constexpr StringLiteral k_ObjectTypeTag = "ObjectType";
+static constexpr StringLiteral k_DataStructureTag = "DataStructure";
+
+static constexpr StringLiteral k_VertexListIdTag = "VertexListId";
+static constexpr StringLiteral k_SharedVertexListTag = "SharedVertexList";
+static constexpr StringLiteral k_VertexSizesIdTag = "VertexSizesId";
```

Comment:
These should be `inline` not `static`.

Also maybe these should be in their own HDF5 constants header.
```suggestion
inline constexpr StringLiteral k_DataTypeTag = "DataType";

inline constexpr StringLiteral k_DataStoreTag = "DataStore";
inline constexpr StringLiteral k_TupleShapeTag = "TupleShape";
inline constexpr StringLiteral k_ComponentShapeTag = "ComponentShape";
inline constexpr StringLiteral k_DataObjectIdTag = "DataObjectId";
inline constexpr StringLiteral k_DataArrayTag = "DataArray";

inline constexpr StringLiteral k_ObjectTypeTag = "ObjectType";
inline constexpr StringLiteral k_DataStructureTag = "DataStructure";

inline constexpr StringLiteral k_VertexListIdTag = "VertexListId";
inline constexpr StringLiteral k_SharedVertexListTag = "SharedVertexList";
inline constexpr StringLiteral k_VertexSizesIdTag = "VertexSizesId";
```